### PR TITLE
Upgrade freestyle to 0.4.0 and dependencies. Product cartesian in endpoints

### LINKED ISSUE #14: Update to Freestyle 0.4.0

### LINKED ISSUE #18: Use cartesian product to build endpoints
This piece of code does not need either for comprehension or monads. The actions are sequential and they don't depends for the others.

```scala 
def build(
      streamMetrics: Stream[Task, WebSocketFrame],
      fromClient: Sink[Task, WebSocketFrame]): FS.Seq[HttpService] = {
    for {
      healthEndpoint   <- healthcheck
      protoEndpoint    <- protoMetric
      metricsWebsocket <- websocketMetrics(streamMetrics, fromClient)
    } yield healthEndpoint |+| metricsWebsocket |+| protoEndpoint
  }
```

This could be the result of implement the cartesian product.

```scala
def build(
      streamMetrics: Stream[Task, WebSocketFrame],
      fromClient: Sink[Task, WebSocketFrame]): FS[HttpService] =
    (healthcheck, protoMetric, websocketMetrics(streamMetrics, fromClient)).mapN(_ |+| _ |+| _)
```

We are going to need the cat version 1.0-MF1, so for this reason, we will have to implement it after upgrade the project to Freestyle 0.4.0 in the issue https://github.com/frees-io/freestyle-opscenter/issues/14

### PR DESCRIPTION
**Pull request by eperinan**
It resolves #18 and resolves #14.

In this PR I was to update Freestyle to 0.4.0 and Http4s to 0.18.0

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,22 +1,22 @@
+import sbtorgpolicies.OrgPoliciesPlugin.autoImport.%%
```
Comment (juanpedromoreno) [inline build.sbt]: It shouldn't be needed

File: build.sbt [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
+
 pgpPassphrase := Some(getEnvVar("PGP_PASSPHRASE").getOrElse("").toCharArray)
 pgpPublicRing := file(s"$gpgFolder/pubring.gpg")
 pgpSecretRing := file(s"$gpgFolder/secring.gpg")
 
 resolvers += Resolver.sonatypeRepo("snapshots")
+resolvers += Resolver.sonatypeRepo("releases")
+
+addCompilerPlugin("org.scalamacros" % "paradise" % "2.1.0" cross CrossVersion.full)
```
Comment (juanpedromoreno) [inline build.sbt:8]: Are we sure we need this compiler plugin in this place?

**Review by juanpedromoreno — APPROVED** [review]
Code LGTM, good job.

As side note, I left a couple of questions/comments.

File: build.sbt (hunk 1 above)
Comment (eperinan) [reply build.sbt]: That import was generated by IntelliJ IDEA. I just removed in the next commit.

File: build.sbt (hunk 2 above)
Comment (eperinan) [reply build.sbt:8]: I tried to put in the file `plugins.sbt` but it didn't work. After discuss it with @fedefernandez, he told me that the place for that line is `build.sbt`.

**Pull request merged**